ankihub/gui/subdeck_due_date_dialog.py

```
@@ -0,0 +1,217 @@
+            "<strong>• Keep as is:</strong> Leave the subdeck and notes unchanged.<br>"
+            "<strong>• Set new due date:</strong> Pick a new date to be reminded later."
+        )
+        options_label = QLabel(options_text)
+        options_label.setWordWrap(True)
+        main_layout.addWidget(options_label)
+
+        # Add stretch to push buttons to bottom
+        main_layout.addStretch()
+
+        # Buttons layout
+        button_layout = QHBoxLayout()
+        button_layout.setSpacing(12)
+
+        # Set new due date button (leftmost)
+        self.set_new_date_button = QPushButton("Set new due date")
+        self.set_new_date_button.clicked.connect(self._on_set_new_due_date)  # type: ignore[attr-defined]
+        button_layout.addWidget(self.set_new_date_button)
+
+        # Keep as is button (middle)
+        self.keep_as_is_button = QPushButton("Keep it")
+        self.keep_as_is_button.clicked.connect(self._on_keep_as_is)  # type: ignore[attr-defined]
+        button_layout.addWidget(self.keep_as_is_button)
+
+        # Move to main deck button (rightmost, primary action)
+        self.move_to_main_button = QPushButton("Move to main deck")
+        self.move_to_main_button.clicked.connect(self._on_move_to_main_deck)  # type: ignore[attr-defined]
+        self.move_to_main_button.setDefault(True)
+        button_layout.addWidget(self.move_to_main_button)
+
+        main_layout.addLayout(button_layout)
+
+    def _on_move_to_main_deck(self):
+        """Handle moving subdeck to main deck."""
+        try:
+            success = move_subdeck_to_main_deck(self.subdeck_config)
+            if success:
+                tooltip(f"'{self.subdeck_name}' moved to main deck", parent=aqt.mw)
+                self.accept()
+                aqt.mw.deckBrowser.refresh()
+            else:
+                showInfo("Failed to move subdeck to main deck. Please try again.", parent=aqt.mw)
```

:wrench: I don't think trying again will help much here, if `move_subdeck_to_main_deck` fails. It's probably better to just have an exception that propagates to the exception handler, which will then open the `ErrorDialog`.

Instead of returning a boolean from `move_subdeck_to_main_deck`, it can raise an exception.